src/Simulator/FloodyFluidSimulator.cpp

```diff
@@ -111,9 +111,11 @@ void cFloodyFluidSimulator::SimulateBlock(cChunk * a_Chunk, int a_RelX, int a_Re
 		
 		// If source creation is on, check for it here:
 		if (
-			(m_NumNeighborsForSource > 0) &&  // Source creation is on
-			(MyMeta == m_Falloff) &&          // Only exactly one block away from a source (fast bail-out)
-			!IsPassableForFluid(Below) &&     // Only exactly 1 block deep
+			(m_NumNeighborsForSource > 0) &&    // Source creation is on
+			(MyMeta == m_Falloff) &&            // Only exactly one block away from a source (fast bail-out)
+				(!IsPassableForFluid(Below) ||  // Only exactly 1 block deep
+				(Below == E_BLOCK_STATIONARY_WATER)  // Or a source block underneath
```

Comment:
Do you mean, when you activate Lava source block createion?
It wasn't defined, but I see your point. I've overseen the m_StationaryFluid variable. Will be fixed.